simulator/infra/replacement/cache_replacement.cpp

```diff
@@ -68,10 +71,120 @@ std::size_t LRUCacheInfo::update()
     return lru_elem;
 }
 
+//////////////////////////////////////////////////////////////////
+
+struct LRU_tree_node
+{
+    explicit LRU_tree_node(int new_way_number = 0, std::size_t new_lru_branch_ptr = 0)
```

Comment:
By the way, why way identifier is integer here? It is `size_t` everywhere else...

Reply:
To be fair, because `core::tree<>` doesn't allow to share 2 identical nodes. In old implementation with more functional (at least for this issue) tree container it was `size_t`. But comprehension between the notes is made by this parameter (in order to find ways in the tree), so I've decided it to be negative while it is not in a leaf of the tree.

Reply:
OK. Than you have to add few more `narrow_cast` since x64 MSVC has sizeof(int) == 4 and sizeof(size_t) == 8.